apps/files/src/mixins.js

```diff
@@ -163,6 +163,48 @@ export default {
         }
       }
     },
+    $_ocUpload_addDirectoryToQue (e) {
+      let files = e.target.files || e.dataTransfer.files
+      if (!files.length) return
+
+      let basePath = this.$route.params.item === undefined ? '/' : this.$route.params.item + '/'
+      let path = basePath
+      for (let file of files) {
+        let folders = file.webkitRelativePath.replace('/' + file.name, '')
+        folders = folders.split('/')
+
+        for (let folder in folders) {
+          console.log(this.files)
+          let folderName = folders[folder]
+          let exists = this.files.find((n) => {
+            if (n['name'] === folderName) {
+              return n
+            }
+          })
+          if (!exists) {
+            this.$client.files.createFolder(path + folderName)
```

Comment:
any further processing (uploading content into the newly created folder) should be done async (.then())